pkg/generate/app/cmd/resolve.go

```diff
@@ -511,7 +525,18 @@ func AddMissingComponentsToRefBuilder(
 				input.Use(repo)
 				input.ExpectToBuild = true
 				repo.UsedBy(input)
-				repo.BuildWithDocker()
+				repo.SetStrategy(generate.StrategyDocker)
+				return input
+			})
+			result = append(result, refs...)
+
+		case info.Jenkinsfile && (g.Strategy == generate.StrategyUnspecified || g.Strategy == generate.StrategyPipeline):
+			refs := b.AddComponents([]string{"scratch"}, func(input *app.ComponentInput) app.ComponentReference {
+				input.Resolver = app.FirstMatchResolver{Searcher: app.DockerClientSearcher{}}
```

Comment:
There may well be a better way to do this.  I found the concept of a "scratch" component which doesn't seem to be much used, but was useful here because the code expects every repo to be attached to a component of some sort.  In the other strategy cases the component to use is clear; here I'm using "scratch" as a fake component.

I set the resolver to be fairly minimal but able to respond to the request for "scratch".  I wasn't clear that there was a neater way of doing this without a resolver but I'm all ears.